src/mugs.js

```diff
@@ -620,6 +621,10 @@ define([
                 // Is this necessary, since itext is a reference?
                 // It probably triggers handlers.
                 this.p[propertyPath] = itext;
+                this.form.fire({
+                    type: 'change',
+                    mug: _this
+                });
```

Comment:
I think this will fire a duplicate change event if `unlink` is true ([mugs.js:79-100](https://github.com/dimagi/Vellum/blob/save-button-fix/src/mugs.js#L79-L100)), and omit other parts of the typical change process (the property change logic is kind of a complicated/messy: [form.js:661](https://github.com/dimagi/Vellum/blob/save-button-fix/src/form.js#L661)) if `unlink` is false. What do you think of this implementation (replaces lines 618-627):
```js
} else if (itext.id !== id) {
    // HACK ensure that the property really changes so change handlers fire.
    this.__data[propertyPath] = null;
}
itext.id = id;
this.p[propertyPath] = itext;
```